### PR TITLE
Add daily build non-windows platforms

### PR DESCRIPTION
**Pull request by TravisEz13**
## PR Summary

Add daily build non-windows platforms
  - Also, make the [Feature] tag work in VSTS for non-windows
  - Also, add a way to force feature tests to run
  - Also, fix an issue where -workingdirectory didn't work when running async 
   

Branched from #7658 as the windows CI is taking a long time to work out the test issues

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by TravisEz13** [comment]
Linux build with `FORCE_FEATURE=True` https://powershell.visualstudio.com/PowerShell/_build/results?buildId=1245&view=logs

**Comment by TravisEz13** [comment]
macos build with `FORCE_FEATURE=True` https://powershell.visualstudio.com/PowerShell/_build/results?buildId=1246

**Comment by iSazonov** [comment]
@TravisEz13 Have you already considered running the tests with a robot like CoreFX team does? This is more convenient than add `[Feature]`.

File: tools/travis.ps1 [hunk 1]
```diff
@@ -154,30 +154,45 @@ function Set-DailyBuildBadge
...
 # only update the build badge for the cron job
-$cronBuild = $env:TRAVIS_EVENT_TYPE -eq 'cron'
+$cronBuild = $env:TRAVIS_EVENT_TYPE -eq 'cron' -or $env:BUILD_REASON -eq 'Schedule'
 $isFullBuild = $isDailyBuild -or $hasFeatureTag
 
 if($Stage -eq 'Bootstrap')
 {
+    if($cronBuild -and $env:TF_BUILD)
+    {
+        Write-Host "##vso[build.updatebuildnumber]Daily-$env:BUILD_SOURCEBRANCHNAME-$env:BUILD_SOURCEVERSION-$((get-date).ToString("yyyyMMddhhss"))"
```
Comment (adityapatwardhan) [inline tools/travis.ps1]: the format should be `yyyyMMddhhmm` ss is for seconds.

File: tools/travis.ps1 [hunk 2]
```diff
@@ -154,30 +154,45 @@ function Set-DailyBuildBadge
...
 }
 else
 {
     $commitMessage = $env:TRAVIS_COMMIT_MESSAGE
 }
 
 # Run a full build if the build was trigger via cron, api or the commit message contains `[Feature]`
-$hasFeatureTag = $commitMessage -match '\[feature\]'
-$hasPackageTag = $commitMessage -match '\[package\]'
+$hasFeatureTag = $commitMessage -match '\[feature\]' -or $env:FORCE_FEATURE -eq 'True'
```
Comment (adityapatwardhan) [inline tools/travis.ps1]: Could you add an comment about `FORCE_FEATURE` and `FORCE_PACKAGE`

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

**Comment by TravisEz13** [comment]
@adityapatwardhan your issues have been addressed as well as codefactor and the test issues

File: .vsts-ci/windows.yml [hunk 3]
```diff
@@ -20,7 +20,12 @@ resources:
   clean: true
 
 steps:
-  - powershell:  Write-Host "##vso[build.updatebuildnumber]$env:BUILD_SOURCEBRANCHNAME-$env:BUILD_SOURCEVERSION-$((get-date).ToString("yyyyMMddhhss"))"
+  - powershell: |
+      Get-ChildItem -Path env:
+    displayName: Capture envirenment
```
Comment (adityapatwardhan) [inline .vsts-ci/windows.yml]: typo environment

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/Executor.cs [hunk 4]
```diff
@@ -194,6 +206,13 @@ internal void ExecuteCommandAsyncHelper(Pipeline tempPipeline, out Exception exc
                 tempPipeline.Error.DataReady += new EventHandler(ErrorObjectStreamHandler);
                 PipelineFinishedWaitHandle waiterThereIsAFlyInMySoup = new PipelineFinishedWaitHandle(tempPipeline);
```
Comment (adityapatwardhan) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/Executor.cs]: Please change name of the variable.

File: tools/travis.ps1 [hunk 5]
```diff
@@ -154,30 +154,62 @@ function Set-DailyBuildBadge
...
+    $commitId = $null
+    if ($env:TRAVIS_EVENT_TYPE)
+    {
+        # We are in Travis-CI
+        $commitId = $env:TRAVIS_PULL_REQUEST_SHA
+
+        # If the current job is a pull request, the env variable 'TRAVIS_PULL_REQUEST_SHA' contains
+        # the commit SHA of the HEAD commit of the PR.
+        $commitMessage = git log --format=%B -n 1 $commitId
+        Write-Verbose "commitMessage: $commitMessage" -verbose
```
Comment (adityapatwardhan) [inline tools/travis.ps1]: Should Write-Log be used here instead?

File: tools/travis.ps1 [hunk 6]
```diff
@@ -154,30 +154,62 @@ function Set-DailyBuildBadge
...
 }
 else
 {
     $commitMessage = $env:TRAVIS_COMMIT_MESSAGE
 }
 
 # Run a full build if the build was trigger via cron, api or the commit message contains `[Feature]`
-$hasFeatureTag = $commitMessage -match '\[feature\]'
-$hasPackageTag = $commitMessage -match '\[package\]'
+# or the environment valiable `FORCE_FEATURE` equals `True`
```
Comment (adityapatwardhan) [inline tools/travis.ps1]: typo variable

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

**Review by adityapatwardhan — APPROVED** [review]

**Review by SteveL-MSFT — APPROVED** [review]

**Pull request merged**